lesson14.html

```diff
@@ -0,0 +1,172 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Tooltips</title>
+    <style>
+      .container_blue {
+        display: flex;
+        flex-direction: row;
+        gap: 50px;
+        padding: 20px;
+        background-color: white;
+        border: 5px solid blue;
+        width: 90%;
+        opacity: 0;
+        animation: slideIn 1s forwards 1s;
+      }
+      .trigger {
+        background-color: white;
+        padding: 10px;
+        border: 2px solid red;
+        position: relative;
+        display: inline-block;
+      }
+      @keyframes slideIn {
+        0% {
+          transform: translateX(-100%);
+          opacity: 0;
+        }
+        100% {
+          transform: translateX(0);
+          opacity: 1;
+        }
+      }
+      .tttgeneral {
+        visibility: hidden;
+        opacity: 0;
+        transition: opacity 1s ease, visibility 1s ease;
+        background-color: gray;
+        color: black;
+        text-align: center;
+        border-radius: 9px;
+        position: absolute;
+        z-index: 1;
+      }
+      .tooltip1 .tooltiptext1 {
+        top: 50%;
+        width: 120px;
+        padding: 5px 10px;
+        left: 0;
+        transform: translate(-15%, -140%);
+      }
+      .tooltip1:hover .tooltiptext1 {
+        visibility: visible;
+        opacity: 1;
+      }
+      .tooltip1 .tooltiptext1::after {
+        content: " ";
```

Comment:
try to extract all common styles into generic selections. And keep in the specific selectors only the unique styles

Reply:
ex: "content" and "position" are common for all ::after elements